From a46145038e1146b09d0c3c3636f57400b67fd53d Mon Sep 17 00:00:00 2001
From: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Date: Mon, 15 Dec 2025 14:43:17 +0200
Subject: [PATCH] s32n79: env: Disable SMMU and enable APS MRU msi-controller

Make sure SMMU is disabled and APS MRU is enabled as msi-controller,
in order to have a functional NETC IP and networking capabilities.

Signed-off-by: Andrei Cherechesu <andrei.cherechesu@nxp.com>
---
 include/configs/s32n79.h | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/include/configs/s32n79.h b/include/configs/s32n79.h
index 0d8a9cf01b2..2d52c24a9e7 100644
--- a/include/configs/s32n79.h
+++ b/include/configs/s32n79.h
@@ -35,7 +35,10 @@
 #define SCSI_PART_ROOTFS		"1:2"
 #define SCSI_ROOT			"/dev/sdb2"
 
-#define FDT_FIXUP ";"
+#define FDT_FIXUP \
+	"fdt addr ${fdt_addr}; fdt resize; " \
+	"fdt set /cis_subsys/iommu status \"disabled\"; " \
+	"fdt set /aps_subsys/msi-controller status \"okay\";"
 
 #if defined(CONFIG_XEN_SUPPORT)
 #  define XEN_COMMON_ENV \
-- 
2.45.2

